presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -935,6 +958,77 @@ public synchronized void rollback()
         }
     }
 
+    public void beginQuery(ConnectorSession session)
+    {
+        String queryId = session.getQueryId();
+        HiveIdentity identity = new HiveIdentity(session);
+
+        synchronized (this) {
+            checkState(
+                    !currentQueryId.isPresent() && !hiveTransactionSupplier.isPresent(),
+                    "Query already begun: %s while starting query %s",
+                    currentQueryId,
+                    queryId);
+            currentQueryId = Optional.of(queryId);
+        }
+
+        hiveTransactionSupplier = Optional.of(Suppliers.memoize(() ->
+        {
+            long heartbeatInterval = configuredTransactionHeartbeatInterval.map(Duration::toMillis).orElse(getServerExpectedHeartbeatIntervalMillis());
+            long transactionId = delegate.openTransaction(identity);
+            log.debug("Using hive transaction %s for query %s", transactionId, queryId);
+
+            ScheduledFuture<?> heartbeatTask = heartbeatExecutor.scheduleAtFixedRate(
+                    () -> delegate.sendTransactionHeartbeat(identity, transactionId),
+                    0,
+                    heartbeatInterval,
+                    MILLISECONDS);
+
+            return new HiveTransaction(identity, queryId, transactionId, heartbeatTask);
+        }));
+    }
+
+    private long getServerExpectedHeartbeatIntervalMillis()
+    {
+        String hiveServerTransactionTimeout = delegate.getConfigValue(HIVE_TXN_TIMEOUT.toString()).orElse("300s");
```

Comment:
`HIVE_TXN_TIMEOUT` (`hive.txn.timeout`) is deprecated in favor of `MetastoreConf.TXN_TIMEOUT` (`metastore.txn.timeout`), so you're likely defaulting to Hive defaults rather than metastore's configuration.

Reply:
Utility methods around MetastoreConf are not very user friendly, have made the changes for MetastoreConf